Plugins/Geant4/src/Geant4Converters.cpp

```diff
@@ -0,0 +1,327 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Geant4/Geant4Converters.hpp"
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Material/HomogeneousSurfaceMaterial.hpp"
+#include "Acts/Material/Material.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "Acts/Surfaces/CylinderBounds.hpp"
+#include "Acts/Surfaces/CylinderSurface.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Surfaces/RadialBounds.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Surfaces/TrapezoidBounds.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+
+#include "G4Box.hh"
+#include "G4LogicalVolume.hh"
+#include "G4Material.hh"
+#include "G4Trap.hh"
+#include "G4Trd.hh"
+#include "G4Tubs.hh"
+#include "G4VPhysicalVolume.hh"
+#include "G4VSolid.hh"
+
+Acts::Transform3 Acts::Geant4AlgebraConverter::transform(
+    const G4ThreeVector& g4Trans) {
+  Transform3 gTransform = Transform3::Identity();
+  Vector3 scaledTrans =
+      Vector3(scale * g4Trans[0], scale * g4Trans[1], scale * g4Trans[2]);
+  gTransform.pretranslate(scaledTrans);
+  return gTransform;
+}
+
+Acts::Transform3 Acts::Geant4AlgebraConverter::transform(
+    const G4RotationMatrix& g4Rot, const G4ThreeVector& g4Trans) {
+  // Create the translation
+  Vector3 translation(scale * g4Trans[0], scale * g4Trans[1],
+                      scale * g4Trans[2]);
+  // And the rotation to it
+  RotationMatrix3 rotation;
+  rotation << g4Rot.xx(), g4Rot.yx(), g4Rot.zx(), g4Rot.xy(), g4Rot.yy(),
+      g4Rot.zy(), g4Rot.xz(), g4Rot.yz(), g4Rot.zz();
+  Transform3 transform = Transform3::Identity();
+  transform.matrix().block(0, 0, 3, 1) = rotation.col(0);
+  transform.matrix().block(0, 1, 3, 1) = rotation.col(1);
+  transform.matrix().block(0, 2, 3, 1) = rotation.col(2);
+  transform.matrix().block(0, 3, 3, 1) = translation;
+  return transform;
+}
+
+Acts::Transform3 Acts::Geant4AlgebraConverter::transform(
+    const G4Transform3D& g4Trf) {
+  // Create the translation
+  Vector3 translation(scale * g4Trf.dx(), scale * g4Trf.dy(),
+                      scale * g4Trf.dz());
+  // And the rotation to it
+  RotationMatrix3 rotation;
+  rotation << g4Trf.xx(), g4Trf.yx(), g4Trf.zx(), g4Trf.xy(), g4Trf.yy(),
+      g4Trf.zy(), g4Trf.xz(), g4Trf.yz(), g4Trf.zz();
+  Transform3 transform = Transform3::Identity();
+  transform.matrix().block(0, 0, 3, 1) = rotation.col(0);
+  transform.matrix().block(0, 1, 3, 1) = rotation.col(1);
+  transform.matrix().block(0, 2, 3, 1) = rotation.col(2);
+  transform.matrix().block(0, 3, 3, 1) = translation;
+  return transform;
+}
```

Comment:
could this be implemented using `Acts::Transform3 Acts::Geant4AlgebraConverter::transform(const G4RotationMatrix& g4Rot, const G4ThreeVector& g4Trans)` or vice versa?

Reply:
Indeed it can.